src/qibolab/instruments/qm.py

```diff
@@ -29,6 +30,57 @@
 )
 from qibolab.sweeper import Parameter
 
+# TODO: Split the contents of this module to multiple files
+
+PortId = Tuple[str, int]
+"""Type for port definition, for example: ("con1", 2)."""
+IQPortId = Union[Tuple[PortId], Tuple[PortId, PortId]]
+"""Type for collections of IQ ports."""
+
+
+@dataclass
+class QMPort(Port):
+    controller_i: PortId
+    controller_q: Optional[PortId] = None
+
+    # TODO: Shall we implement seperate setters for i and q?
+    _offset: float = 0.0
+    _gain: int = 0
+    _filter: Optional[Dict[str, float]] = None
+
+    def __iter__(self):
+        yield self.controller_i
+        if self.controller_q is not None:
+            yield self.controller_q
+
+    @property
+    def offset(self):
+        return self._offset
+
+    @offset.setter
+    def offset(self, value):
+        self._offset = value
```

Comment:
When things are happening like this, in case it's working, I'd adopt the same solution proposed for dummy: directly expose the attribute.